src/awkward/operations/ak_from_arrow.py

```diff
@@ -68,4 +68,10 @@ def _impl(array, generate_bitmasks, highlevel, behavior):
             if awkwardarrow_type.record_is_scalar:
                 out = out._getitem_at(0)
 
+    def remove_revertable(layout, **kwargs):
+        if hasattr(layout, "__pyarrow_original"):
+            del layout.__pyarrow_original
+
+    out.recursively_apply(remove_revertable)
```

Comment:
We don't let an array get into the users' hands with a bunch of `__pyarrow_original` attributes in it.

Reply:
It's a hack for this particular function, `ak.from_arrow`, and it doesn't extend beyond it. This manual removal is like garbage collection: we could try something slicker but then we might remove too much or too little—a final run that clears the private attribute from the whole tree is the simplest, cleanest way to be sure.